dds/idl/marshal_generator.cpp

```diff
@@ -1630,6 +1630,46 @@ namespace {
     return bounded;
   }
 
+  ExtensibilityKind max_extensibility_kind(AST_Type* type, bool root = true)
+  {
+    static ExtensibilityKind max_exten;
+    static std::vector<AST_Type*> type_stack;
+
+    if (root) {
+      type_stack.clear();
+      max_exten = extensibilitykind_final;
+    }
+
+    type = resolveActualType(type);
+    for (size_t i = 0; i < type_stack.size(); ++i) {
+      // If we encounter the same type recursively, then it has already been considered
+      if (type == type_stack[i]) return max_exten;
+    }
+    type_stack.push_back(type);
+
+    const Classification fld_cls = classify(type);
+    if (fld_cls & CL_STRUCTURE || fld_cls & CL_UNION) {
+      ExtensibilityKind this_exten = be_global->extensibility(type);
+      if (this_exten == extensibilitykind_mutable) {
+        return extensibilitykind_mutable;
+      }
+      if (max_exten < this_exten) max_exten = this_exten;
+
+      const Fields fields(fld_cls & CL_STRUCTURE ? dynamic_cast<AST_Structure*>(type) : dynamic_cast<AST_Union*>(type));
+      const Fields::Iterator fields_end = fields.end();
+      for (Fields::Iterator i = fields.begin(); i != fields_end; ++i) {
+        if (extensibilitykind_mutable == max_extensibility_kind(type, false)) {
+          return extensibilitykind_mutable;
+        }
+      }
+      return max_exten;
+    } else if (fld_cls & CL_ARRAY || fld_cls & CL_SEQUENCE) {
+      return max_extensibility_kind(type, false);
```

Comment:
Need to use base type. As it is it will go into a recursive loop because `type` will always be the same.

```suggestion
    } else if (fld_cls & CL_ARRAY) {
      return max_extensibility_kind(dynamic_cast<AST_Array*>(type)->base_type(), false);
    } else if (fld_cls & CL_SEQUENCE) {
      return max_extensibility_kind(dynamic_cast<AST_Sequence*>(type)->base_type(), false);
```